staging/src/k8s.io/client-go/metadata/metadata.go

```diff
@@ -191,7 +191,7 @@ func (c *client) Get(ctx context.Context, name string, opts metav1.GetOptions, s
 	}
 	obj, err := result.Get()
 	if runtime.IsNotRegisteredError(err) {
-		klog.V(5).Infof("Unable to retrieve PartialObjectMetadata: %#v", err)
+		klog.FromContext(ctx).V(5).Info("Could not retrieve PartialObjectMetadata", "err", err)
```

Comment:
The unit test(staging/src/k8s.io/client-go/metadata/metadata_test.go) corresponding to the `Get` method can use `_, ctx := ktesting.NewTestContext(t)` to pass context to test the contextual logging.

Reply:
Thank you, should be resolved with 04e5522099623d66c1536ebc4253528280381c99

Please take another look

```
make test WHAT=./staging/src/k8s.io/client-go/metadata/ GOFLAGS="-v"
+++ [1208 10:21:40] Set GOMAXPROCS automatically to 12
+++ [1208 10:21:40] Running tests without code coverage and with -race
=== RUN   TestClient
=== RUN   TestClient/GET_is_able_to_convert_a_JSON_object_to_PartialObjectMetadata
    metadata.go:194: I1208 10:05:11.876937] Could not retrieve PartialObjectMetadata err="no kind \"Pod\" is registered for version \"v1\" in scheme \"k8s.io/apimachinery/pkg/apis/meta/internalversion/scheme/register.go:27\""
=== RUN   TestClient/LIST_is_able_to_convert_a_JSON_object_to_PartialObjectMetadata
    metadata.go:230: I1208 10:05:11.878102] Could not retrieve PartialObjectMetadataList err="no kind \"PodList\" is registered for version \"v1\" in scheme \"k8s.io/apimachinery/pkg/apis/meta/internalversion/scheme/register.go:27\""
=== RUN   TestClient/GET_fails_if_the_object_is_JSON_and_has_no_kind
=== RUN   TestClient/GET_fails_if_the_object_is_JSON_and_has_no_apiVersion
=== RUN   TestClient/GET_fails_if_the_object_is_JSON_and_not_clearly_metadata
    metadata.go:194: I1208 10:05:11.880156] Could not retrieve PartialObjectMetadata err="no kind \"Pod\" is registered for version \"v1\" in scheme \"k8s.io/apimachinery/pkg/apis/meta/internalversion/scheme/register.go:27\""
=== RUN   TestClient/Delete_fails_if_DeleteOptions_cannot_be_serialized_to_JSON
=== RUN   TestClient/DeleteCollection_fails_if_DeleteOptions_cannot_be_serialized_to_JSON
--- PASS: TestClient (0.01s)
    --- PASS: TestClient/GET_is_able_to_convert_a_JSON_object_to_PartialObjectMetadata (0.01s)
    --- PASS: TestClient/LIST_is_able_to_convert_a_JSON_object_to_PartialObjectMetadata (0.00s)
    --- PASS: TestClient/GET_fails_if_the_object_is_JSON_and_has_no_kind (0.00s)
    --- PASS: TestClient/GET_fails_if_the_object_is_JSON_and_has_no_apiVersion (0.00s)
    --- PASS: TestClient/GET_fails_if_the_object_is_JSON_and_not_clearly_metadata (0.00s)
    --- PASS: TestClient/Delete_fails_if_DeleteOptions_cannot_be_serialized_to_JSON (0.00s)
    --- PASS: TestClient/DeleteCollection_fails_if_DeleteOptions_cannot_be_serialized_to_JSON (0.00s)
PASS
ok      k8s.io/kubernetes/staging/src/k8s.io/client-go/metadata (cached)
```